src/manifolds/Segre.jl

```diff
@@ -0,0 +1,465 @@
+@doc raw"""
+    Segre{𝔽,V} <: AbstractManifold{𝔽}
+
+The Segre manifold
+````math
+    \mathcal{S} = \operatorname{Seg}(𝔽^{n_1} \times \dots \times 𝔽^{n_d})
+````
+is the set of rank-one tensors in ``𝔽^{n_1} \otimes \dots \otimes 𝔽^{n_d}``.
+
+When ``𝔽 = ℝ``, the Segre manifold is a normal Riemannian covering of
+````math
+    \mathcal{P} = ℝ^{+} \times S^{n_1 - 1} \times \dots \times S^{n_d - 1}
+````
+with the [warped product metric](https://en.wikipedia.org/wiki/Warped_product) [`inner`](@ref inner(::Segre, ::Any)). The tuple ``(n_1, \dots, n_d)`` is called the _valence_ of the manifold.
+
+The geometry is summarized in [JacobssonSwijsenVandervekenVannieuwenhoven:2024](@cite).
+
```

Comment:
Side track: I usually love nerdy details. Could we finish this doc strip with a sentence why this is called the Segre(-Veronese) manifold? Maybe with a reference to a paper and a wikipedia link to the one/two mathematicians(?) that came up with this?